### PR TITLE
Build static lib too

### PR DESCRIPTION
**Pull request by davidbrochart**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

@conda-forge-admin, please rerender

### CODE REVIEW

**Comment by davidbrochart** [comment]
@xhochy ok to merge?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,26 +13,58 @@ source:
...
+    test:
+      commands:
+        - test -f $PREFIX/lib/libaws-c-common${SHLIB_EXT}  # [unix]
+        - test ! -f $PREFIX/lib/libaws-c-common.a  # [unix]
+        - test -f $PREFIX/include/aws/common/config.h  # [unix]
+        - test -f $PREFIX/lib/cmake/AwsCFlags.cmake  # [unix]
+        # Check for the non-existence of -Werror in the CFLAGS
+        - "! grep Werror $PREFIX/lib/cmake/AwsCFlags.cmake"  # [unix]
+        - if not exist %PREFIX%\\Library\\bin\\aws-c-common.dll exit 1  # [win]
+        - if exist %PREFIX%\\Library\\lib\\aws-c-common.lib exit 1  # [win]
```
Comment (xhochy) [inline recipe/meta.yaml:42]: This should exist. You need a thin static library on Windows even for dynamic linkage.

Comment (davidbrochart) [reply recipe/meta.yaml:42]: So, since the tests pass, it means we have a `.dll` but not a `.lib`. Not sure why?

Comment (xhochy) [reply recipe/meta.yaml:42]: Because of a weird `conda-build` behaviour not running part of the build.

**Comment by davidbrochart** [comment]
Thanks @xhochy!

**Comment by xhochy** [comment]
The variants cannot be used at the same time as they would clobber eachother. Either the static one should depend on the shared one or there should be a run_constrained <0a0 on each other.

**Comment by davidbrochart** [comment]
Shouldn't we test for `include/aws/common/config.h`? I see we don't do it in https://github.com/conda-forge/s2n-feedstock/blob/master/recipe/meta.yaml.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -13,26 +13,60 @@ source:
...
+    requirements:
+      build:
+        - cmake
+        - {{ compiler('c') }}
+        - ninja
+      host:
+        - {{ pin_subpackage("aws-c-common", exact=True) }}
+    test:
+      commands:
+        - test ! -f $PREFIX/lib/libaws-c-common${SHLIB_EXT}  # [unix]
```
Comment (xhochy) [inline recipe/meta.yaml:60]: It exists as we depend on the shared lib.
```suggestion
```

**Comment by xhochy** [comment]
> Shouldn't we test for `include/aws/common/config.h`? I see we don't do it in https://github.com/conda-forge/s2n-feedstock/blob/master/recipe/meta.yaml.

Yes, we should test for that in the main package.

**Comment by xhochy** [comment]
I'm confused why now the `run_exports` is not working correctly 🤔

**Comment by xhochy** [comment]
Closing as stale.

**Pull request closed without merging**